sync.go

```diff
@@ -249,17 +262,18 @@ func (mw *MultiWallet) RestartSpvSync() error {
 }
 
 func (mw *MultiWallet) CancelSync() {
-	if mw.syncData.cancelSync != nil {
+	if mw.IsSyncing() {
```

Comment:
This is not quite correct. Syncing may have completed but this method is called to disconnect the wallet from the network backend. The actual problem here is that `mw.syncData.cancelSync` may not be nil (even after initial sync completes) but `mw.syncData.addressDiscoveryCompletedOrCanceled` would be accessible because `mw.syncData.activeSyncData` would be nil and `mw.syncData.addressDiscoveryCompletedOrCanceled` is actually `mw.syncData.activeSyncData.addressDiscoveryCompletedOrCanceled`.